freqtrade/freqai/RL/BaseReinforcementLearningModel.py

```diff
@@ -318,8 +317,24 @@ def build_ohlc_price_dataframes(self, data_dictionary: dict,
         prices_test.rename(columns=rename_dict, inplace=True)
         prices_test.reset_index(drop=True)
 
+        train_df = self.drop_ohlc_from_df(train_df, dk)
+        test_df = self.drop_ohlc_from_df(test_df, dk)
+
         return prices_train, prices_test
 
+    def drop_ohlc_from_df(self, df: DataFrame, dk: FreqaiDataKitchen):
+        """
+        Given a dataframe, drop the ohlc data
+        """
+        drop_list = ['%-raw_open', '%-raw_low', '%-raw_high', '%-raw_close']
+
+        if self.rl_config["drop_ohlc_from_features"]:
+            df.drop(drop_list, axis=1, inplace=True)
+            feature_list = dk.training_features_list
+            feature_list = [e for e in feature_list if e not in drop_list]
```

Comment:
I'm not sure i understand the point of this (maybe it's a subtle bug?).

`feature_list` is a local variable at this point - which is not updated but reassigned (so it'll not mutate the original `dk.training_features_list`).
I doubt that's the intend ... ?

Reply:
Not a bug. Some people want normalized ohlc features in their training and others don’t. This helps the users who “don’t.”

I believe you are incorrect on this point. `feature_list` points to `dk.training_feature_list`. That is good, we expect that in this current usage. 

If you did *not* want feature_list to point to dk.training_feature_list, you’d need to do 

`feature_list = dk.training_feature_list[:]`

Or use `deepcopy`.

Reply:
at the start of the line - you're correct.

After this line however, feature_list is a new, independent list.

you could also write it as follows - and it would be identical (it would just not reuse/reassign the feature_list variable.

```
feature_list_new = [e for e in feature_list if e not in drop_list]
``` 

My main point is - `dk.training_features_list` will come out unmodified of this method - which i don't think is intended.

Reply:
Ok, you are right, I missed this piece.

I reviewed it, and technically the algorithm works weather we update `dk.training_feature_list` or not. To be clear, we are updating it here explicitly.

Reply:
i'm not arguing that - updating `dk.training_feature_list` or not is both fine for me (whatever's necessary for the logic to work) - but the code didn't do what it intended (it didn't actualy write back to `dk.training_feature_list`) - which makes it explicit - but an implicit bug that might be hard to find SHOULD it cause problems.